src/components/Register/Register.tsx

```diff
@@ -0,0 +1,597 @@
+  .object({
+    id: yup.string().required().min(9).max(12),
+    email: yup.string().email().required(),
+    password: yup
+      .string()
+      .required()
+      .min(8)
+      .matches(/^\S*$/, 'Mật khẩu không được có khoảng trắng'),
+    gender: yup.string().required(),
+    date: yup.string().nullable().required('Date is required field'),
+    province_id: yup.number().min(1, 'Thông tin không được để trống'),
+    district_id: yup
+      .number()
+      .required()
+      .min(1, 'Thông tin không được để trống'),
+    ward_id: yup.number().required().min(1, 'Thông tin không được để trống')
+  })
+  .required();
+
+interface Ward {
+  id: number;
+  name: string;
+  district_id: number;
+}
+
+interface District {
+  id: number;
+  name: string;
+  wards: Ward[];
+  province_id: number;
+}
+interface Province {
+  id: number;
+  name: string;
+  districts: District[];
+}
+
+const provinces: Province[] = [
+  {
+    id: 1,
+    name: 'Hà Nội',
+    districts: [
+      {
+        id: 1,
+        name: 'Hoàn Kiếm',
+        province_id: 1,
+        wards: [
+          {
+            id: 1,
+            name: 'Chương Dương',
+            district_id: 1
+          },
+          {
+            id: 2,
+            name: 'Cửa Nam',
+            district_id: 1
+          }
+        ]
+      },
+      {
+        id: 2,
+        name: 'Hà Đông',
+        province_id: 1,
+        wards: [
+          {
+            id: 1,
+            name: 'La Khê',
+            district_id: 2
+          },
+          {
+            id: 2,
+            name: 'Nguyễn Trãi',
+            district_id: 2
+          }
+        ]
+      }
+    ]
+  },
+  {
+    id: 2,
+    name: 'Hồ Chí Minh',
+    districts: [
+      {
+        id: 1,
+        name: 'Quận 1',
+        province_id: 2,
+        wards: [
+          {
+            id: 1,
+            name: 'Bến Nghé',
+            district_id: 1
+          },
+          {
+            id: 2,
+            name: 'Bến Thành',
+            district_id: 1
+          }
+        ]
+      },
+      {
+        id: 2,
+        name: 'Quận 2',
+        province_id: 2,
+        wards: [
+          {
+            id: 1,
+            name: 'An Khánh',
+            district_id: 2
+          },
+          {
+            id: 2,
+            name: 'An Phú',
+            district_id: 2
+          }
+        ]
+      }
+    ]
+  }
+];
+const Register = () => {
+  const {
+    control,
+    handleSubmit,
+    formState: { isValid },
+    setValue,
+    watch
+  } = useForm<FormData>({
+    defaultValues,
+    mode: 'onChange',
+    resolver: yupResolver(schema)
+  });
+
+  const provinceId = watch('province_id');
+  const districtId = watch('district_id');
+
+  useEffect(() => {
+    const subscription = watch((value, { name, type }) => {
+      if (name === 'province_id') {
+        setValue('district_id', 0);
+        setValue('ward_id', 0);
+      }
+      if (name === 'district_id') {
+        setValue('ward_id', 0);
+      }
```

Comment:
```suggestion
      if (name === 'province_id') {
        setValue('district_id', '');
        setValue('ward_id', '');
      }
      if (name === 'district_id') {
        setValue('ward_id', '');
      }
```

Để support defaultValue="" cho Select